src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -396,20 +395,16 @@ def create_compile_context(self, target, target_workdir):
 
   def execute(self):
     requested_compiler = JvmPlatform.global_instance().get_options().compiler
-    if requested_compiler != self.compiler_name:
-      return
-    deprecated_conditional(
-      lambda:  requested_compiler == self.Compiler.ZINC,
-      removal_version='1.20.0.dev0',
-      entity_description='Requested a deprecated compiler: [{}].'.format(requested_compiler),
-      hint_message='Compiler will be defaulted to [{}].'.format(self.compiler_name))
-
-    if requested_compiler == self.Compiler.ZINC and self.compiler_name == self.Compiler.RSC:
-      # Issue a deprecation warning (above) and rewrite zinc to rsc, as zinc is being deprecated.
-      JvmPlatform.global_instance().get_options().compiler = RankedValue(0, self.compiler_name)
+    if requested_compiler == self.Compiler.ZINC.value and self.compiler_name == self.Compiler.RSC.value:
```

Comment:
This could be `if self.Compiler(requested_compiler) == self.Compiler.ZINC`, for one, but the jvm platform option should really be using `type=cls.Compiler` in the `register()` method in `jvm_platform.py`.  I would really like to clean up the enum usage for these options by registering with the enum type before merging, as this has already caused confusion.

Reply:
> I would really like to clean up the enum usage for these options by registering with the enum type before merging, as this has already caused confusion.

It's arguably caused confusion exactly due to drive by reviews suggesting its usage, rather than making those kinds of changes in isolated reviews. But I will do it here.

Reply:
> drive by reviews suggesting its usage

I'm not sure I agree with this characterization. From my perspective, I wasn't aware that `self.Compiler` was an `enum` at all, and then I saw this issue in which a simple equality comparison was made into a complicated conditional. Simply changing that conditional to wrap the `requested_compiler` in an `enum` instead of unwrapping multiple `.value` fields seemed appropriate.

The PR description specifically states:
> was not triggering a deprecation warning due to enum/str comparison mixup

This is something that is *specifically* checked against for `enum`s: https://github.com/pantsbuild/pants/blob/d45af2806326ee71c8f163daea6c0512b58fadd2/src/python/pants/util/objects.py#L337-L338

If the use of an `enum` here isn't appropriate, then don't introduce a new one. I was trying to advise how to more easily use the `enum` type, while referencing previous work I had done to solve exactly the problems in this PR. I don't understand how your response corresponds to this.

Reply:
Apologies. Since there had been previous attempts to use an enum here, I assumed that this usage was related. Sorry about that.